docs/architecture/adr-071-crypto-v2-multi-curve.md

```diff
@@ -0,0 +1,597 @@
+}
+```
+
+
+##### Keyring
+
+The new `Keyring` interface will serve as a central hub for managing and fetching `CryptoProviders`. To ensure a smoother migration path, the new Keyring will be backward compatible with the previous version. Since this will be the main API from which applications will obtain their `CryptoProvider` instances, the proposal is to extend the Keyring interface to include the methods:
+
+
+```go
+type Keyring interface {
+  // methods from Keyring/v1
+  
+  // ListCryptoProviders returns a list of all the stored CryptoProvider metadata.
+  ListCryptoProviders() ([]ProviderMetadata, error)
+  
+  // GetCryptoProvider retrieves a specific CryptoProvider by its id.
+  GetCryptoProvider(id string) (CryptoProvider, error)
+}
+```
+
+##### Especial use case: remote signers
+
+It's important to note that the `CryptoProvider` interface is versatile enough to be implemented as a remote signer. This capability allows for the integration of remote cryptographic operations, which can be particularly useful in distributed or cloud-based environments where local cryptographic resources are limited or need to be managed centrally.
+
+Here are a few of the services that can be leveraged:
+
+* AWS CloudHSM
+* Azure Key Vault
+* HashiCorp Vault
+* Google Cloud KMS
+
+## Alternatives
+
+It is important to note that all the code presented in this document is not in its final form and could be subject to changes at the time of implementation. The examples and implementations discussed should be interpreted as alternatives, providing a conceptual framework rather than definitive solutions. This flexibility allows for adjustments based on further insights, technical evaluations, or changing requirements as development progresses.
+
+## Decision
+
+We will:
+
+* Refactor the module structure as described above.
+* Define types and interfaces as the code attached.
+* Refactor existing code into new structure and interfaces.
+* Implement Unit Tests to ensure no backward compatibility issues.
+
+## Consequences
+
+### Impact on the SDK codebase
+
+We can divide the impact of this ADR into two main categories: state machine code and client related code.
+
+#### Client
+
+The major impact will be on the client side, where the current `Keyring` interface will be replaced by the new `Keyring` interface.
+This means that any piece of code that makes use of the `Record` struct will need to be adapted to use the new `CryptoProvider` instead.
+This will also affect a large number of unit tests that will need to be adapted/replaced with new ones.
+
+#### State Machine
+
+The impact on the state machine code will be minimal, the modules affected (at the time of writing this ADR)
+are the `x/accounts` module, specifically the `Authenticate` function and the `x/auth/ante` module. These function will need to be adapted to use a `CryptoProvider` service to make use of the `Verifier` instance.
```

Comment:
why is accounts affected? the authenticate does not assume the signer is a key, how would this change?

Reply:
At first sight the answer should be the same as a [previous one ](https://github.com/cosmos/cosmos-sdk/pull/18824#discussion_r1440402149), but again maybe after 6 months of that writing the `auth` module had significant changes and might not have a great impact.

Reply:
x/accounts decouples authentication detail from the state-machine, for instance authentication in account can be dealt with using this strategy https://github.com/cosmos/cosmos-sdk/blob/main/x/accounts/defaults/base/account.go#L48

but this is just one